### PR TITLE
ShipStats Rework

### PR DESCRIPTION
**Pull request by iTTou**
- Add ability to open multiple builds of the same ship at the same time
- Change chips into shipSelector to proper list
- Ability to assign multiple builds to selected ships before opening stats
- Build string in the home page now accepts also links and short links
- Add build string input dialog with string validation for both links and build strings (short links not supported)
- Add a bunch of missing localizations
- Add build name to shipStats page

 Must review together with "Add viewmodels caching to shipstats" PR

### CODE REVIEW

File: WoWsShipBuilder.Web/Components/ShipStatsContainer.razor [hunk 1]
```diff
@@ -59,9 +59,9 @@
                         </MudStack>
                         @if (ViewModel.NextShips is not null)
                         {
-                            @foreach (var ship in ViewModel.NextShips)
+                            foreach (var ship in ViewModel.NextShips)
                             {
-                                <MudStack Class="px-2 cursor-pointer" @onmouseup="@(args => UpdateToShip(args, ViewModel.CurrentShipIndex, ship))" AlignItems="AlignItems.Center" id="@(ViewModel.CurrentShipIndex + "_" + ship.Index)">
+                                <MudStack Class="px-2 cursor-pointer" @onmouseup="@(args => UpdateToShip(args, CurrentTabId, ship))" AlignItems="AlignItems.Center" id="@(ViewModel.CurrentShipIndex + "_" + ship.Index)">
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Components/ShipStatsContainer.razor:64]: Use the key property for elements generated in loops

File: WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+    private void Cancel()
+    {
+        MudDialog.Cancel();
+    }
+
+    private void OnConfirm()
+    {
+        string buildStr;
+        if (buildString.Contains("build="))
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor:41]: The variable name "buildString" implies that it only contains the unescaped build string

File: WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        }
+        else
+        {
+            buildStr = buildString;
+        }
+        
+        MudDialog.Close(buildStr.Trim());
+    }
+
+    private string? ValidateBuildString(string buildStr)
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor:54]: Add localization for validation results

File: WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        validBuild = false;
+        if (string.IsNullOrWhiteSpace(buildStr))
+        {
+            return null;
+        }
+        
+        if (buildStr.Contains("share.wowssb.com"))
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor:62]: Do not hardcode urls or hostnames, use data from config file.
Also, why is this not supported? You can retrieve the real url by sending a HTTP request with FollowRedirects disabled

File: WoWsShipBuilder.Web/Pages/Index.razor [hunk 5]
```diff
@@ -91,17 +101,45 @@
 
     private void OnLoadBuild()
     {
-        try
+        if (buildString.Contains("share.wowssb.com"))
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Pages/Index.razor:104]: Again, do not hardcode urls or hostnames unless their structure is defined in code in the same project (like the name of a specific page)

File: WoWsShipBuilder.Web/Pages/ShipStats.razor [hunk 6]
```diff
@@ -29,88 +44,104 @@
...
-        NavManager.TryGetQueryString("build", out string buildString);
-        if (!string.IsNullOrWhiteSpace(ships))
+        NavManager.TryGetQueryString("shipIndexes", out shipIndexesFromUrl);
+        NavManager.TryGetQueryString("build", out shipBuildFromUrl);
+        if (!string.IsNullOrEmpty(shipIndexesFromUrl))
         {
-            string[] indexes = ships.Split(",");
-            shipBuildMapper[indexes[0]] = buildString;
-            AddNewIndexes(indexes);
+            var indexes = shipIndexesFromUrl.Split(",").ToList();
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Pages/ShipStats.razor:70]: Why the .ToList()?

**Review by floribe2000 — CHANGES_REQUESTED** [review]
Overall good changes, however, there are some issues in both code and styling. Please have a look at them.

File: WoWsShipBuilder.Web/Components/ShipStatsContainer.razor [hunk 7]
```diff
@@ -34,15 +34,15 @@
 {
     <MudGrid>
         <MudItem xs="12">
-            <MudPaper Outlined="true" Class="pa-1">
+            <MudPaper Outlined="true" Class="py-2 px-15">
```
[the remark this answers is not in the document]
Comment (iTTou) [reply WoWsShipBuilder.Web/Components/ShipStatsContainer.razor:37]: After the addition of the text field for the build name the padding wasn't centering the items properly anymore so I had to change it out with that

File: WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                return inputBuildValidationResultCache;
+            }
+        }
+
+        if (QueryHelpers.ParseQuery(buildStr).TryGetValue("build", out var buildStringFromUrl))
+        {
+            buildStr = buildStringFromUrl.ToString();
+        }
+
+        if (buildStr.Contains(';'))
```
Comment (floribe2000) [inline WoWsShipBuilder.Web/Dialogs/BuildStringInputDialog.razor:88]: Why is this check necessary? The try-catch should already catch issues if the string is malformed and there is no longer handling for the old case where there was an url-escaped string

**Review by floribe2000 — CHANGES_REQUESTED** [review]
This looks pretty good now, there is just one minor question I had, see the comment. Once this is solved and the merge conflicts are resolved, this can then be merged.

**Pull request merged**